ui/qt-lib/src/Ariadne/UI/Qt/Widgets/WalletTree.hs

```diff
@@ -81,17 +83,27 @@ initWalletTree langFace itemModel selectionModel = do
   return (widget, WalletTree{..})
 
 addWalletClicked :: UiLangFace -> WalletTree -> Bool -> IO ()
-addWalletClicked UiLangFace{..} WalletTree{..} _checked =
+addWalletClicked langFace@UiLangFace{..} walletTree@WalletTree{..} _checked =
   runNewWallet >>= \case
     NewWalletCanceled -> return ()
-    NewWalletAccepted NewWalletParameters{..} -> void $ langPutUiCommand $
+    NewWalletAccepted NewWalletParameters{..} ->
       case nwSpecifier of
-        NewWalletName -> UiNewWallet nwName nwPassword
-        NewWalletMnemonic mnemonic full -> UiRestoreWallet nwName nwPassword mnemonic full
+        NewWalletName -> doCreateNewWallet langFace walletTree nwName nwPassword
+        NewWalletMnemonic mnemonic full -> void $ langPutUiCommand $
+          UiRestoreWallet nwName nwPassword mnemonic full
+
+doCreateNewWallet :: UiLangFace -> WalletTree -> Text -> Maybe Text -> IO ()
+doCreateNewWallet UiLangFace{..} WalletTree{..} name password = do
+  let entropySize = 16 -- FIXME Hardcoded for now, will have to fetch from wallet config
```

Comment:
For that we need to expose wallet configuration to the ui. At this step I could've hardcoded entropy size from the config into `generateMnemonic` in `Main.hs` (or can I?), but in the future I believe we want to preselect default option in the dialog, so configuration should be exposed anyway.

Reply:
I suppose we need to pass not only `MnemonicGenerator`, but also `entropySize :: Byte` or `:: IO Byte` to UI creation function. `:: Byte` should be enough now, `IO Byte` will be needed if we make it possible to update configuration in runtime.